### PR TITLE
Promises added

### PR DESCRIPTION
**Pull request by vyshakhbabji**
Promises added

### CODE REVIEW

File: src/phoneLine.js [hunk 1]
```diff
@@ -451,45 +451,54 @@ PhoneLine.prototype.cancel = function() {
...
+                        return data;
+                    });
+            }
         }
-    }
-    else {
-        return Promise.reject(new Error('No line or no active line'));
-    }
+        else {
+         return    Promise.reject(new Error('No line or no active line'));
```
Comment (kirill-konshin) [inline src/phoneLine.js]: Should be just a throw statement.

**Comment by vyshakhbabji** [comment]
added throw statements

File: build/ringcentral-web-phone.js [hunk 2]
```diff
@@ -12589,70 +12677,101 @@ PhoneLine.prototype.getSession = function() {
 /*--------------------------------------------------------------------------------------------------------------------*/
 
 PhoneLine.prototype.cancel = function() {
-    var session = this.getSession();
-    session.terminate({statusCode: 486});
-    return Promise.resolve(null);
+    return new Promise(function(resolve, reject) {
+        var session = this.getSession();
```
Comment (kirill-konshin) [inline build/ringcentral-web-phone.js]: this will point not to PhoneLine instance here

**Comment by vyshakhbabji** [comment]
Changed cancel().

File: src/phoneLine.js [hunk 3]
```diff
@@ -520,19 +531,22 @@ PhoneLine.prototype.sendDTMF = function(value, duration) {
...
 
 PhoneLine.prototype.sendInfoDTMF = function(value, duration) {
-    duration = parseInt(duration) || 1000;
-    var session = this.session;
-    session.dtmf(value.toString(), {
-        duration: duration
+
+    return new Promise(function(resolve, reject) {
+        duration = parseInt(duration) || 1000;
+        var session = this.session;
```
Comment (kirill-konshin) [inline src/phoneLine.js]: Same here about `this`

**Comment by vyshakhbabji** [comment]
Promise for infoDTMF

**Pull request merged**